dotnet/src/SemanticKernel/TemplateEngine/PromptTemplateEngine.cs

```diff
@@ -24,86 +23,77 @@ namespace Microsoft.SemanticKernel.TemplateEngine;
 /// </summary>
 public class PromptTemplateEngine : IPromptTemplateEngine
 {
+    private readonly ILogger _log;
+
+    private readonly TemplateTokenizer _tokenizer;
+
     public PromptTemplateEngine(ILogger? log = null)
     {
         this._log = log ?? NullLogger.Instance;
+        this._tokenizer = new TemplateTokenizer(this._log);
     }
 
-    /// <summary>
-    /// Given a prompt template string, extract all the blocks (text, variables, function calls)
-    /// </summary>
-    /// <param name="templateText">Prompt template (see skprompt.txt files)</param>
-    /// <param name="validate">Whether to validate the blocks syntax, or just return the blocks found, which could contain invalid code</param>
-    /// <returns>A list of all the blocks, ie the template tokenized in text, variables and function calls</returns>
+    /// <inheritdoc/>
     public IList<Block> ExtractBlocks(string? templateText, bool validate = true)
     {
         this._log.LogTrace("Extracting blocks from template: {0}", templateText);
-        var blocks = this.TokenizeInternal(templateText);
-        if (validate) { ValidateBlocksSyntax(blocks); }
+        var blocks = this._tokenizer.Tokenize(templateText);
+
+        if (validate)
+        {
+            foreach (var block in blocks)
+            {
+                if (!block.IsValid(out var error))
+                {
+                    throw new TemplateException(TemplateException.ErrorCodes.SyntaxError, error);
+                }
+            }
+        }
 
         return blocks;
     }
 
-    /// <summary>
-    /// Given a prompt template, replace the variables with their values and execute the functions replacing their
-    /// reference with the function result.
-    /// </summary>
-    /// <param name="templateText">Prompt template (see skprompt.txt files)</param>
-    /// <param name="executionContext">Access into the current kernel execution context</param>
-    /// <returns>The prompt template ready to be used for an AI request</returns>
-    public async Task<string> RenderAsync(
-        string templateText,
-        SKContext executionContext)
+    /// <inheritdoc/>
+    public async Task<string> RenderAsync(string templateText, SKContext context)
     {
         this._log.LogTrace("Rendering string template: {0}", templateText);
         var blocks = this.ExtractBlocks(templateText);
-        return await this.RenderAsync(blocks, executionContext);
+        return await this.RenderAsync(blocks, context);
```

Comment:
May be out of scope of this review, but I'm seeing first hand and getting reports from folks that SK behaves differently than the playground with identical prompts and it seems to come down to '\r\n' vs '\n' - I'm wondering if we should go ahead and make that part of the engine to convert everything for consistency?